src/Middleware/SubstituteBindings.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace LaravelDoctrine\ORM\Middleware;
+
+use Closure;
+use Doctrine\Common\Persistence\Proxy;
+use Illuminate\Contracts\Routing\Registrar;
+use Illuminate\Routing\Middleware\SubstituteBindings as IllumintateSubstituteBindings;
+use LaravelDoctrine\ORM\IlluminateRegistry;
+use ReflectionFunction;
+use ReflectionMethod;
+
+class SubstituteBindings extends IllumintateSubstituteBindings
+{
+    /**
+     * @var IlluminateRegistry
+     */
+    protected $registry;
+
+    /**
+     * Create a new bindings substitutor.
+     *
+     * @param Registrar          $router
+     * @param IlluminateRegistry $registry
+     */
+    public function __construct(Registrar $router, IlluminateRegistry $registry)
+    {
+        parent::__construct($router);
+
+        $this->registry = $registry;
+    }
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @param  \Closure                 $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        $this->router->substituteBindings($route = $request->route());
+
+        $this->router->substituteImplicitBindings($route);
+
+        $this->substituteImplicitBindings($route);
+
+        return $next($request);
+    }
+
+    /**
+     * Substitute the implicit Doctrine entity bindings for the route.
+     *
+     * @param  \Illuminate\Routing\Route $route
+     * @return void
+     */
+    protected function substituteImplicitBindings($route)
+    {
+        $parameters = $route->parameters();
+
+        $action = $route->getAction();
+
+        foreach ($this->getParameters($action['uses']) as $parameter) {
+            $class = $parameter->getClass()->getName();
+
+            // Try to find the entity manager for the given class.
+            if (is_null($entityManager = $this->getDoctrineEntityManagerByClass($class))) {
+                continue;
+            }
+
+            if (array_key_exists($parameter->name, $parameters)) {
+                $entity = $entityManager->find($class, $parameters[$parameter->name]);
```

Comment:
Seems they have something in place for both situations.
https://github.com/laravel/framework/blob/5.3/src/Illuminate/Routing/Router.php#L805

Can you try and make it behave it the same way as Eloquent does.